sdk/openai/Azure.AI.OpenAI/tests/ChatToolsTests.cs

```diff
@@ -0,0 +1,221 @@
+{
+    public ChatToolsTests(bool isAsync)
+        : base(Scenario.ChatTools, isAsync)//, RecordedTestMode.Live)
+    {
+    }
+
+    [RecordedTest]
+    [TestCase(Service.Azure)]
+    [TestCase(Service.NonAzure)]
+    public async Task SimpleFunctionToolWorks(Service serviceTarget)
+    {
+        OpenAIClient client = GetTestClient(serviceTarget);
+        string deploymentOrModelName = GetDeploymentOrModelName(serviceTarget);
+
+        var requestOptions = new ChatCompletionsOptions()
+        {
+            DeploymentName = deploymentOrModelName,
+            Tools = { s_futureTemperatureTool },
+            Messages =
+            {
+                new ChatRequestSystemMessage("You are a helpful assistant."),
+                new ChatRequestUserMessage("What should I wear in Honolulu next Thursday?"),
+            },
+            MaxTokens = 512,
+        };
+
+        Response<ChatCompletions> response = await client.GetChatCompletionsAsync(requestOptions);
+        Assert.That(response, Is.Not.Null);
+
+        Assert.That(response.Value, Is.Not.Null);
+        Assert.That(response.Value.Choices, Is.Not.Null.Or.Empty);
+
+        ChatChoice choice = response.Value.Choices[0];
+        Assert.That(choice.FinishReason, Is.EqualTo(CompletionsFinishReason.ToolCalls));
+
+        ChatResponseMessage message = choice.Message;
+        Assert.That(message.Role, Is.EqualTo(ChatRole.Assistant));
+        Assert.That(message.Content, Is.Null.Or.Empty);
+        Assert.That(message.ToolCalls, Is.Not.Null.Or.Empty);
+        Assert.That(message.ToolCalls.Count, Is.EqualTo(1));
+        ChatCompletionsFunctionToolCall functionToolCall = message.ToolCalls[0] as ChatCompletionsFunctionToolCall;
+        Assert.That(functionToolCall, Is.Not.Null);
+        Assert.That(functionToolCall.Function.Name, Is.EqualTo(s_futureTemperatureFunction.Name));
+        Assert.That(functionToolCall.Function.Arguments, Is.Not.Null.Or.Empty);
+
+        ChatCompletionsOptions followupOptions = new()
+        {
+            DeploymentName = deploymentOrModelName,
+            Tools = { s_futureTemperatureTool },
+            MaxTokens = 512,
+        };
+        // Include all original messages
+        foreach (ChatRequestMessage originalMessage in requestOptions.Messages)
+        {
+            followupOptions.Messages.Add(originalMessage);
+        }
+        // And the tool call message just received back from the assistant
+        followupOptions.Messages.Add(new ChatRequestAssistantMessage(choice.Message.Content)
+        {
+            ToolCalls = { functionToolCall },
+        });
+        // And also the tool message that resolves the tool call
+        followupOptions.Messages.Add(new ChatRequestToolMessage(
+            toolCallId: functionToolCall.Id,
+            content: JsonSerializer.Serialize(
+                new
+                {
+                    Temperature = "31",
+                    Unit = "celsius",
```

Comment:
Correct me if I am wrong, but you might've mixed up your functions here. You'd have to return something with `LocationName` and `Date`, AFAICT.

Reply:
I am wondering, where is the argument passed with which the LLM wants us to perform the tool call.

Reply:
I am not following how you agree with this structure with the LLM. While debugging the LLM just decides to also ask for field (`unit` in celcius) which I didn't advertise in the original request. This is a bit confusing

Reply:
It seems that the `ChatRequestToolMessage` doesn't need to be structured data, just FYI. I just verified that.